File: typescript/src/approve.ts
import { exchange, signHash } from "./client";

const MAX_FEE = "1%";

async function main() {
  const res = await exchange({
    action: { type: "approveBuilderFee", maxFeeRate: MAX_FEE },
  });
  const sig = await signHash(res.hash);

  await exchange({
    action: { type: "approveBuilderFee", maxFeeRate: MAX_FEE },
    nonce: res.nonce,
    signature: sig,
  });

  console.log("Builder fee approved.");
}

main();

File: python/approve.py
"""Approve builder fee (one-time setup)."""

from client import exchange, sign_hash

MAX_FEE = "1%"

res = exchange({
    "action": {"type": "approveBuilderFee", "maxFeeRate": MAX_FEE},
})
sig = sign_hash(res["hash"])

exchange({
    "action": {"type": "approveBuilderFee", "maxFeeRate": MAX_FEE},
    "nonce": res["nonce"],
    "signature": sig,
})

print("Builder fee approved.")
